### PR TITLE
Progress to v2.0.0 (Neovim v0.8 Release)

### LINKED ISSUE #391: Preparation for v0.8
This is just a placeholder for keeping up with changes that are coming with Neovim v0.8.

## TODO

- [x] `resolved_capabilities` deprecated (https://github.com/neovim/neovim/pull/17814)
- [x] `nvim_cmd` API addition (https://github.com/neovim/neovim/pull/18366)
- [x] Use `feline` to set up `winbar` for breadcrumbs in active window and filename for inactive window (https://github.com/neovim/neovim/pull/18562)
- [x] Investigate `cmdheight = 0` (https://github.com/neovim/neovim/pull/16251)
- [x] Remove deprecated `filetype.lua` options (https://github.com/neovim/neovim/pull/19216)
- [x] https://github.com/neovim/neovim/pull/19238

### LINKED ISSUE #795: Replace ncdu and htop with cross-platfom equivalents for Windows
**Is your feature request related to a problem? Please describe.**
image
Most of the optional requirements almost work in Windows, except for `ncdu` and `htop`. I would like to be able to replace both of them with cross-platform equivalents that work with Windows.

**Describe the solution you'd like**
For `ncdu`, [gdu](https://github.com/dundee/gdu) seems to be the best choice. It looks and functions like ncdu:
image

For htop, I'm not sure. I personally use https://github.com/ClementTsang/bottom, but there are many other options as well.

**Additional context**

### PR DESCRIPTION
**Pull request by mehalter**
This is a heavy WIP just to keep on top of breaking changes on nightly as they come up. This branch should be considered unstable and in development

This now uses Heirline for the status bar and winbar which is a breaking change

**Version Requirement**: ≥`NVIM v0.8.0`

### TODO

- [x] Add API documentation for `core/utils/lsp.lua`
- [x] Finish API documentation for `core/utils/status.lua`
- [x] Add `mason-null-ls` support (https://github.com/jayp0521/mason-null-ls.nvim)
  - Need to wait for latest PRs to be merged to be feature complete and then testing is necessary
- [x] Finalize new install instructions for stable AstroNvim and make stable the default
  - [x] move nightly development to a dedicated branch
  - [x] make nightly channel use the nightly branch by default
  - [x] update contribution guide to indicate PRs should be merged to nightly
- [x] Get updated documentation ready for all breaking changes
- [x] Verify we have updated to all new Neovim 0.8 APIs and features
- [x] Make sure all plugins have been updated to support Neovim 0.8

Resolves #699
Resolves #391
Resolves #795

### CODE REVIEW

File: .github/README.md [hunk 1]
```diff
@@ -54,8 +54,8 @@ AstroNvim is an aesthetic and feature-rich neovim config that is extensible and
 - Optional Requirements:
   - [ripgrep](https://github.com/BurntSushi/ripgrep) - live grep telescope search (`<leader>fw`)
   - [lazygit](https://github.com/jesseduffield/lazygit) - git ui toggle terminal (`<leader>tl` or `<leader>gg`)
-  - [NCDU](https://dev.yorhel.nl/ncdu) - disk usage toggle terminal (`<leader>tu`)
-  - [Htop](https://htop.dev/) - process viewer toggle terminal (`<leader>tt`)
+  - [go DiskUsage()](https://github.com/dundee/gdu) - disk usage toggle terminal (`<leader>tu`)
+  - [gotop](https://github.com/xxxserxxx/gotop) - process viewer toggle terminal (`<leader>tt`)
```
Comment (sitiom) [inline .github/README.md:58]: So you really do prefer gotop, huh 🤔

**Review by sitiom — CHANGES_REQUESTED** [review]

File: .github/README.md (hunk 1 above)
Comment (mehalter) [reply .github/README.md:58]: gotop has been around for a long time and has many many years of development and support. I think for this use case I would like to stick with something a little bit more future proof

Comment (sitiom) [reply .github/README.md:58]: It's not like `bottom` is unstable or something, and in terms of age, they only have a gap of over a year ( [gotop](https://github.com/xxxserxxx/gotop/commit/40775db60b90cd290a206108cc4d22b236be9ba5) / [bottom](https://github.com/ClementTsang/bottom/commit/0cfcf323a1dba844758e01314b923d0d8665554a) )

Comment (sitiom) [reply .github/README.md:58]: @mehalter I have tested gotop in Windows, and I am experiencing major slowdowns, making it unusable. I am also experiencing this error: https://github.com/xxxserxxx/gotop/issues/4

Comment (mehalter) [reply .github/README.md:58]: okie, i'll replace it with bottom @sitiom

**Comment by mehalter** [comment]
Neovim has officially placed a tentative due date for [Neovim v0.8 release](https://github.com/neovim/neovim/milestone/32). It will be September 30, 2022. I have placed the same due date on the AstroNvim v2.0 milestone in hopes that we can get the update to the `nightly` channel of AstroNvim the day that the new update comes out!

**Pull request merged**